libs/libc/machine/arch_atomic.c

```diff
@@ -28,7 +28,33 @@
 
 #include <stdbool.h>
 #include <stdint.h>
-#include <nuttx/spinlock.h>
+#include <nuttx/irq.h>
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+#ifdef CONFIG_SMP
```

Comment:
hardware must support atomic operation in SMP, so we.don't.need this.codd.block

Reply:
but some architectures may not fully support all atomic operations, eg: -march=armv7e-m does not support atomic8(64 bit) https://github.com/OpenAMP/libmetal/pull/301

Reply:
OK